### ISSUE TITLE
Malformed override does not warn it is incorrect

### ISSUE THREAD

**Issue by lspackova** [msuchane/cizrna#20]
A typo in an override syntax (e.g. subsystem: instead of subsystems:) doesn't warn there is a problem and just doesn't work. In case there are multiple SSTs, the build fails with a duplicate ID error despite the attempt for an SST override. In other cases, it just silently does nothing, and notes are sorted into an incorrect chapter.

Please add a warning that an override is malformed to prevent these cases - or even better, make the pipeline fail in such a case.

**Comment by msuchane** [comment]
Thanks for reporting this, @lspackova.

The issue was that the configuration parser ignored "unknown" fields by default. I've changed it so that it reports those fields and exits the program with an error. For example:

```
Error: 
   0: Cannot parse the tickets configuration file.
   1: .[37][2].overrides: unknown field `subsystem`, expected one of `doc_type`, `components`, `subsystems` at line 43 column 36

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

This is now released as version 0.20.2.

**Comment by lspackova** [comment]
Excellent, thank you so much @msuchane!
